Build failure of using BUILD_REMOTE=1 and BUILD_JNI=1 options

Error message:
```
make[1]: *** No rule to make target 'build/worker-jni/iri-pearldiver-exlib.o',
needed by 'build/remote-worker'.  Stop.
```

The **Makefile** does not take care of the situation of enabling the build options `BUILD_REMOTE` and `BUILD_JNI=1` at the same time.
Besides fixing it, the CI should add the corresponding building test.

--- a/Makefile
+++ b/Makefile
@@ -152,7 +152,11 @@ OBJS += \
 	remote_common.o \
 	remote_interface.o
 
-WORKER_OBJS := $(addprefix $(OUT)/worker-,$(filter-out remote_interface.o, $(OBJS)))
+WORKER_EXCLUDE_OBJS := remote_interface.o
+ifeq ("$(BUILD_JNI)", "1")
+WORKER_EXCLUDE_OBJS += jni/iri-pearldiver-exlib.o
+endif
+WORKER_OBJS := $(addprefix $(OUT)/worker-,$(filter-out $(WORKER_EXCLUDE_OBJS), $(OBJS)))
 WORKER_CFLAGS := $(filter-out -DENABLE_REMOTE, $(CFLAGS))
 endif
 

